src/main/resources/avro/readmethods.avdl

```diff
@@ -14,8 +14,10 @@ to `ReferenceSet`s containing that same `Reference`. If no reference is
 specified, all `ReadGroup`s must be aligned to the same `ReferenceSet`.
 */
 record SearchReadsRequest {
-  /** If nonempty, restrict this query to reads within the given readgroups. */
-  array<string> readGroupIds = [];
+  /**
+  The ReadGroup to search.
+  */
+  string readGroupId;
```

Comment:
* I think most of the dozen independent changes in this PR are non-controversial, since they don't break _any_ published code, and they do simplify things. @jeromekelleher, if you want to divide and conquer, I think you could get consensus on over half of the PR quickly, while we continue to discuss the remainder. @calbach --  other than this `readGroupIds[]` proposal, do you know which other edits are potentially breaking changes?  

* @diekhans -- I agree the API shouldn't be set in stone; I also think that we shouldn't casually break compatibility -- that's why this is a good discussion. And I strongly agree that we need to build real applications as part of the design process -- I'm delighted to see the work on the new-and-improved reference server, in addition to the earlier work by EBI, NCBI, https://github.com/googlegenomics/api-provider-local-java, and Google's product.

* we discussed `ReadGroup` and `ReadGroupSet` at great length the first time around. There was passionate support (including by @lh3) to allow retrieving from a single `ReadGroup`, and there was passionate support (including by me) to allow retrieving all reads from a single `ReadGroupSet` -- that's what led to the current method accepting an array of `readGroupIds[]`. There's more than one way to satisfy those two requirements, but I believe that any proposal needs to continue to do so.